src/dune_tui/dune_tui.ml

```diff
@@ -144,35 +33,288 @@ module Tui = struct
     fun () -> !(Lazy.force setup)
   ;;
 
-  let start () = ()
+  (** style for diving visual elements like borders or rules *)
+  let divider_attr = A.(fg red)
+
+  (** style for helpful ui elements like scrollbar structures or help text *)
+  let helper_attr = A.(fg yellow)
+
+  (** style for user feedback like message count, or scrollbar position *)
+  let user_feedback_attr = A.(fg cyan)
+
+  (** [horizontal_rule ~attr ~w] draws a horizontal line with [attr] of length [w]. *)
+  let horiztonal_rule ~attr ~w = I.uchar attr (Uchar.of_int 0x2015) w 1
+
+  (** Here we keep some persistent state about the program that we udpate each time we
+      render. This allows other components to "react" to changes using [Lwd]. *)
+
+  let term_size = Lwd.var (Term.size (term ()))
```

Comment:
This should be lazily initialized

Reply:
It will get initialized in the render loop anyway, so I will set it to some dummy value for now.